src/ledger/test/LedgerCloseMetaStreamTests.cpp

```diff
@@ -427,3 +431,68 @@ TEST_CASE("METADATA_DEBUG_LEDGERS works", "[metadebug]")
     }
     REQUIRE(gotToExpectedSize);
 }
+
+TEST_CASE("meta stream contains reasonable meta", "[ledgerclosemeta]")
+{
+
+    TmpDirManager tdm(std::string("metatest-") + binToHex(randomBytes(8)));
+    TmpDir td = tdm.tmpDir("meta-ok");
+    std::string path = td.getName() + "/stream.xdr";
+
+    VirtualClock clock;
+    Config cfg = getTestConfig();
+    cfg.METADATA_OUTPUT_STREAM = path;
+
+    {
+        // Do some stuff
+        using namespace stellar::txtest;
+        auto app = createTestApplication(clock, cfg);
+        auto bal = app->getLedgerManager().getLastMinBalance(2);
+
+        auto root = TestAccount::createRoot(*app);
+
+        // Ledger 2 creates the account, which we do directly so emit no meta
+        // for.
+        auto acc1 = root.create("acc1", bal);
+
+        // Ledger 3 uses closeLedger so emits interesting meta.
+        std::vector<TransactionFrameBasePtr> txs = {
+            root.tx({payment(acc1.getPublicKey(), 1000)})};
+        closeLedger(*app, txs);
+    }
+
+    XDRInputFileStream in;
+    in.open(path);
+    LedgerCloseMeta lcm;
+    uint32_t maxSeq = 0;
+    while (in.readOne(lcm))
+    {
+        if (lcm.v() == 2)
+        {
+            maxSeq = lcm.v2().ledgerHeader.header.ledgerSeq;
+            if (lcm.v2().ledgerHeader.header.ledgerSeq == 3)
+            {
+                REQUIRE(lcm.v2().txProcessing.size() == 1);
```

Comment:
why not just convert to json and do a string comparison of the entire record for "ledger 3"?
That way we'll have in the test the actual expected meta (which will allow us to see when we make any meta changes - as this breaks Horizon).

Also note that vcurrent and vnext have a different format (which we'll see in the test)

Reply:
ok, doing so (it adds a lot of test vector files, I eyeballed them a bit but second opinions on whether they're correct welcome!)